gateway-ha/src/main/java/io/trino/gateway/ha/router/ExternalRoutingGroupSelector.java

```diff
@@ -76,18 +74,16 @@ public class ExternalRoutingGroupSelector
             throw new RuntimeException("Invalid URL provided, using "
                     + "routing group header as default.", e);
         }
-        httpClient = new JettyHttpClient(new HttpClientConfig());
+        this.httpClient = requireNonNull(httpClient, "httpClient is null");
     }
 
     @Override
     public String findRoutingGroup(HttpServletRequest servletRequest)
     {
-        Request request;
-        JsonBodyGenerator<RoutingGroupExternalBody> requestBodyGenerator;
         try {
             RoutingGroupExternalBody requestBody = createRequestBody(servletRequest);
-            requestBodyGenerator = jsonBodyGenerator(ROUTING_GROUP_EXTERNAL_BODY_JSON_CODEC, requestBody);
-            request = preparePost()
+            JsonBodyGenerator<RoutingGroupExternalBody> requestBodyGenerator = jsonBodyGenerator(ROUTING_GROUP_EXTERNAL_BODY_JSON_CODEC, requestBody);
+            Request request = preparePost()
```

Comment:
This change looks unrelated to "Support HttpClient options for external routing group selector". Please separate a commit.

Reply:
Separted to https://github.com/trinodb/trino-gateway/pull/580/commits/fd379cd2184fc740c097b7ed268cb5207cb8d871 commit :) Thx for pointing it out ! Greatly appreciate it.